lib/logstash/plugin_mixins/jdbc/statement_handler.rb

```diff
@@ -148,41 +148,28 @@ def perform_query(db, sql_last_value)
     private
 
     def build_query(db, sql_last_value)
-      @parameters = create_bind_values_hash
-      if statement_prepared.false?
-        prepended = parameters.keys.map{|v| v.to_s.prepend("$").to_sym}
-        @prepared = db[statement, *prepended].prepare(:select, name)
-        statement_prepared.make_true
-      end
       # under the scheduler the Sequel database instance is recreated each time
       # so the previous prepared statements are lost, add back
-      if db.prepared_statement(name).nil?
-        db.set_prepared_statement(name, prepared)
-      end
-      bind_value_sql_last_value(sql_last_value)
-      begin
-        db.call(name, parameters)
-      rescue => e
-        # clear the statement prepared flag - the statement may be closed by this
-        # time.
-        statement_prepared.make_false
-        raise e
-      end
+      prepared = db.prepared_statement(name)
+      prepared ||= db[statement, *positional_bind_placeholders].prepare(:select, name)
```

Comment:
review note: Sequel's `Dataset#prepare` always registers the resulting prepared statement with the `Database` so that it can be looked up with `Database#prepared_statement(name)`.